common/simulation/spi.cpp

```diff
@@ -22,11 +22,16 @@ void sim_spi::SimTMC2130Spi::transmit_receive(
     uint8_t reg = control & ~write_mask;
 
     if (control & write_mask) {
+        // This is a write command.
         register_map[reg] = data;
+    } else {
+        // A read command will return the data from the previous read command.
+        auto out_iter = receive.begin();
+        out_iter =
```

Comment:
It looks like this is returning the address of the register as the first byte, according to the datasheet the first byte of the packet is actually a status byte with some flags. I don't think it'll affect any of the tests though.